crates/turborepo-paths/src/absolute_system_path.rs

```diff
@@ -197,6 +197,45 @@ impl AbsoluteSystemPath {
     pub fn components(&self) -> Components<'_> {
         self.0.components()
     }
+
+    // Produces a path from self to other, which may include directory traversal
+    // tokens. Given that both parameters are absolute, we _should_ always be
+    // able to produce such a path. The exception is when crossing drive letters
+    // on Windows, where no such path is possible. Since a repository is
+    // expected to only reside on a single drive, this shouldn't be an issue.
+    pub fn relative_path_to(&self, other: &Self) -> AnchoredSystemPathBuf {
```

Comment:
If `self` and `other` are the same path, we'd end up with an empty path, will that work okay?